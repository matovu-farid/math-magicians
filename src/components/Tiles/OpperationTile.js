import { Component } from 'react';
import './Tile.css';
import PropTypes from 'prop-types';

export default class OpperationTile extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { opp } = this.props;
    return (<li className="tile orange">{opp}</li>);
  }
}
OpperationTile.propTypes = { opp: PropTypes.string.isRequired };
