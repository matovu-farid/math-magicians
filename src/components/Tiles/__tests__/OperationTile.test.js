import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import OpperationTile from '../OpperationTile';

it('renders correctly', () => {
  const tree = renderer.create(<OpperationTile opp="+" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should display correct value', () => {
  render(<OpperationTile opp="+" />);
  const button = screen.queryByRole('button');

  expect(button).toHaveTextContent('+');
});
