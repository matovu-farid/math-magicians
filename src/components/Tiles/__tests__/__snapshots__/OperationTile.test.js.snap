// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="tile orange"
  onClick={[Function]}
  type="button"
>
  +
</button>
`;
